impl Display for SyncStatus

posvyatokum opened the pull request:
Implement Display for SyncStatus that is only different from Debug in StateSync variant, i.e. it does not write potentially huge HashMap.
Use it in tracing of client.sync_status to remove noise from logs.

nikurt commented on chain/client-primitives/src/types.rs:
```diff
@@ -300,6 +300,17 @@ impl From<SyncStatus> for SyncStatusView {
+impl std::fmt::Display for SyncStatus {
+    fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
+        match self {
+            SyncStatus::StateSync(hash, _) => write!(f, "StateSync {{ hash: {:?} }}", hash),
```
Would it be possible to format `StateSync` with the use of this function (to be made public)?
https://github.com/near/nearcore/blob/a5ede1d1cd54906ac91cbb614028b5323ddee0c0/chain/client/src/client.rs#L2287

nikurt replied:
Maybe the better way if to define `ShardSyncDownload::debug` using `format_shard_sync_phase()`

posvyatokum commented:
I suspect I made things worse, here is a summary:
1. Moved state sync status formatting functions closer to structure definitions to avoid circular dependencies. Now I can use it.
2. Changed accessing element 0 in the formatting function to safe getter, because the localnet test was failing otherwise.
3. I generally listened to @nagisa's comment, but I don't know how to redefine Debug for just one variant and I didn't want to 'print' all variants by hand, so I added extra structure just to define pretty debugging. This prompted little refactoring around client.
4. I didn't find a way to pass # flag in tracing the way we pass ? or %, so changed the debug line that started it all a little.
